docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+  - **Advantage**: Maintaining migrations as well as the data access layer code is simplified.
+
+- **Option 2: Use *separate databases***: While using the same database makes sense from a maintenance point of view it might restrict us in the technology choices since a mobile app and backend service do have completely differnt constraints. It might as well hinder us from optimising our backend service as we have to consider mobile constraints.
+  - **Advantage**: Databases can be optimised according to the platform specific needs.
+
+### What kind of database should be used?
+
+- **Option 1: Use a *relational database***: In the past we have used relational databases, promoting a clear data structure and model. A lot of care has been taken to migrate old records to adhere to data model changes. No data redundancy.
+- **Option 2: Use a *no sql database***: A much more flexible option in terms of data model, with performance prioritised over data integrity. Data redundancy.
+
+
+## Lightning node
+
+Our design principles promote self-custody and self-sovereignty - as a result the users lightning node must run on her phone.
+
+### How should the lightning node run on the phone?
+
+- **Option 1: *Foreground process***: The lightning node will be started only when the app is opened.
+  - **Advantages**: low battery consumption, and simple implementation
+- **Option 2: *Background process***: The lightning node runs as background task on the phone, even if the app is closed.
+  - **Advantages**: payments can be received at any time
+
+Another alternative has been discussed if push notifications could be used to notify the user about an incoming payment asking to accept it by opening the app. We have not addressed the question of privacy.
+
+However, this feature might not be important for the MVP.
+
+### How do we extend the lightning channel for DLCs?
+
+DLCs are not supported by the lightning channels outof the box and a custom extension needs to be implemented. Downwards compatibility should be granted though.
+
+ - **Option 1: *Virtual Channels***: ..
+ - **Option 2: ...**:
+
+## State management
+
+How are state changes managed in 10101.
+
+ - **Option 1: *Events***: State changes are reflected with immutable events allowing to recreate a state by applying all events in the correct order. While this architectural pattern introduces some complexities it allows for a much more flexible, async processing. In itchysats an event driven architecture has been implemented for parts of the solution.
+ - **Option 2: *API***: State changes are promoted through synchronous api calls with a blocking wait to check if the call has been processed successfully. This is a simple paradigm an less complex.  
+
+*Note, these options are not exhaustive nor very general for state management. Also the target architecture could make use of a combination of both approaches. Further elaboration and discussion is required here.* 
+
+## Coordinator API
```

Comment:
I would make a difference between the orderbook and the coordinator.

I'm aiming for a separation of concern from a legal perspective in the long run, meaning, from a technical perspective I wold like to seperate these two things from the getgo to simplify a legal separation later on.

For example, _a way_ how this could play out: 
- orderbook contains orders
- taker grabs an order which includes information of the counter party
- taker _tells_ the coordinator he wants to take this order
- coordinator opens positions with both parties simultaneously.

Reply:
We can build the orderbook api as a separate module, but I would not build it as a separate service from the "get-go". because coordinating multiple micro services is hard, just think of testing, deployment (CI/CD), etc. If we stick to a proper modularization of the components within one service, we can still easily separate them afterwards if we really need to.

Reply:
I actually think it is easier because we can evolve and deploy it independently.

Reply:
Small note: if we are going to build toward a microservice based architecture it may be worth considering some kind of distributed tracing setup again, though we may only want to track very high level spans like REST calls from service to service to reduce the amount of data (an issue we had in itchysats)

Reply:
> Small note: if we are going to build toward a microservice based architecture it may be worth considering some kind of distributed tracing setup again, though we may only want to track very high level spans like REST calls from service to service to reduce the amount of data (an issue we had in itchysats)

That's exactly the kind of complexity i fear if we start building the coordinator and orderbook as separate micro services. We also need to consider retry mechanisms, distributed transactions, etc.

Reply:
> We also need to consider retry mechanisms, distributed transactions, etc.

Eventually yes, but we don't have to solve all problems in distributed computing at day 0.